src/components/Banners/GithubUsernameBanner.tsx

```diff
@@ -0,0 +1,47 @@
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Icon from "@leafygreen-ui/icon";
+import { uiColors } from "@leafygreen-ui/palette";
+import { useLocation } from "react-router-dom";
+import { paths } from "constants/routes";
+import { size } from "constants/tokens";
+import { GetUserSettingsQuery } from "gql/generated/types";
+import { GET_USER_SETTINGS } from "gql/queries";
+
+const { green } = uiColors;
+export const GithubUsernameBanner = () => {
+  const { pathname } = useLocation();
+  const isPatchesPage = pathname.startsWith(paths.user);
+
+  // USER SETTINGS QUERY
+  const { data: userSettingsData } = useQuery<GetUserSettingsQuery>(
+    GET_USER_SETTINGS
```

Comment:
We should avoid doing a query if we don't need to and are not on the correct page.